### ISSUE TITLE
Utilize Jekyll to include all mod data in page request (or as single api call), rather than multiple calls

### ISSUE THREAD

**Issue by wispborne** [OpenGameMods-group/ES-Mod-Share#29]
MZ:

> Someone also ought to take a look at the relative benefits of JSON files vs. Jekyll files vs. just editing a data structure in a Javascript. In any case eventually it would be great if the game only has to pull down a single web page (script or HTML) to get all the information it needs on the available mods.
> 
> My plan for how to do the mods was to embed the metadata in data-\* attributes generated from Jekyll data and embedded directly into the HTML file of the mods gallery, to speed up load times by limiting the number of separate files that have to be retrieved. But I have no idea if that would actually be the best way of doing it or not.

**Comment by wispborne** [comment]
My own thoughts:
I'd like to keep the separate folder structure for mods, it makes them easier to submit, easier to view, easier to modify, everything.
However, Jekyll lets us iterate over folders, which I took advantage of in a recent commit. There's no reason I know of that I couldn't go a step further and use Jekyll to combine all mod data in a single place.

The next question is, should that data be included as part of the initial page response, or reached via a single api call (rather than the n number of calls we do right now), and return all mod data as json?

I see a single api call as the favorable way - this will let the page load very quickly without waiting for all of the mod content to load, making it apparently more responsive (or performance or whatever the word is these days). 
A single api call also serves as a possible way for the actual ES game to grab mod data and display it ingame, which is an item on the timeline. 

Note: at some point it'll also need to support `skip` and `take`, because ain't nobody gonna wait for the data for 150 mods at once.
Also, images still need to load async. Obviously.

**Comment by wispborne** [comment]
This is mostly done. Mod data is now stored as yaml, but still in self-contained folders. All mods are available as json at https://evan153.github.io/ES-Mod-Share/mods-json.html. 

Rather than an http request for every mod, a single request is done, after the page has loaded. In this way, the page still loads very quickly, but it doesn't need to wait for the mod content. In reality, this is probably still slower than just including all of the mod content in the initial load because of compression and stuff, but whatever, it's fast enough, performance really is not an issue. 

I've also laid the basic groundwork for supporting multiple banner images so we can have a pretty carousal. That is, I'm now processing yaml into json, which means that I have a little wiggle room in creating the end product, without having to modify the input. 

The way the html is generated could also _really_ be streamlined. There's no good reason now for it to be pulled down as json and then hacked into existance via javascript; jekyll can just generate the pages directly from yaml with no json needed. That's pretty low priority, though...it's fine for now. Heck, one could argue that as long as I'm exposing it as an API, I should be eating what I'm serving. 

CC @endless-sky 
**tldr: Check out https://evan153.github.io/ES-Mod-Share/mods-json.html**
